velox/experimental/cudf/exec/CudfFilterProject.cpp

```diff
@@ -59,30 +111,106 @@ CudfFilterProject::CudfFilterProject(
           nvtx3::rgb{220, 20, 60}, // Crimson
           operatorId,
           fmt::format("[{}]", project ? project->id() : filter->id())),
-      hasFilter_(info.hasFilter),
+      hasFilter_(filter != nullptr),
+      lazyDereference_(
+          dynamic_cast<const core::LazyDereferenceNode*>(project.get()) !=
+          nullptr),
       project_(project),
       filter_(filter) {
-  resultProjections_ = *(info.resultProjections);
-  identityProjections_ = std::move(identityProjections);
+  VELOX_CHECK(!(lazyDereference_ && filter));
+  if (filter_ != nullptr && project_ != nullptr) {
+    folly::Synchronized<exec::OperatorStats>& opStats = Operator::stats();
+    opStats.withWLock([&](auto& stats) {
+      stats.setStatSplitter(
+          [filterId = filter_->id()](const auto& combinedStats) {
+            return splitStats(combinedStats, filterId);
+          });
+    });
+  }
+}
+
+void CudfFilterProject::initialize() {
+  initializeFilterProject();
+
+  auto info = exprsAndProjection();
   const auto inputType = project_ ? project_->sources()[0]->outputType()
                                   : filter_->sources()[0]->outputType();
 
   // convert to AST
   if (cudfDebugEnabled()) {
     int i = 0;
-    for (auto expr : info.exprs->exprs()) {
+    for (const auto& expr : info.exprs->exprs()) {
       std::cout << "expr[" << i++ << "] " << expr->toString() << std::endl;
       debugPrintTree(expr);
+      ++i;
     }
   }
   std::vector<std::shared_ptr<velox::exec::Expr>> projectExprs;
   if (hasFilter_) {
-    // First expr is Filter, rest are Project
     filterEvaluator_ = ExpressionEvaluator({info.exprs->exprs()[0]}, inputType);
     projectExprs = {info.exprs->exprs().begin() + 1, info.exprs->exprs().end()};
   }
+
   projectEvaluator_ = ExpressionEvaluator(
       hasFilter_ ? projectExprs : info.exprs->exprs(), inputType);
+
+  filter_.reset();
+  project_.reset();
+}
+
+void CudfFilterProject::initializeFilterProject() {
```

Comment:
The contents of this can simply be included in `CudfFilterProject::initialize()`